File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        numset = set()
        maxPath = 1
        # create a numset
        for i in nums:
            numset.add(i)
            
        print(numset)
        for i in numset: 
            path = 1
            while i + 1 in numset:
                path = path + 1
                i = i + 1
            if path > maxPath:
                maxPath = path
        
        return maxPath

            

File: Data Structures & Algorithms/binary-search/submission-1.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int l = 0;
        int r = nums.Length;
        while (l < r) {
            int mid = (r - l)/2 + l;
            if (nums[mid] == target){
                return mid;
            }
            else if (nums[mid] < target) {
                l = mid + 1;
            }
            else {                
                r = mid;
            }
        }
        if (l < nums.Length && nums[l] == target){
            return l;
        }

        return -1;
    }
}
